modular_zubbers/code/datums/quirks/positive_quirks/defensive.dm

```diff
@@ -0,0 +1,37 @@
+/datum/quirk/defensive
+	name = "Jerk"
+	desc = "You are rude and impatient, and want people to get out of your way. "
+	icon = FA_ICON_FROWN_OPEN
+	medical_record_text = "Patient has anger issues."
+	value = 6
+	gain_text = span_notice("You feel like punching someone.")
+	lose_text = span_notice("You feel calmer.")
+	quirk_flags = QUIRK_HIDE_FROM_SCAN
+	COOLDOWN_DECLARE(shovingspam)
+
+/datum/quirk/defensive/add(client/client_source)
+	. = ..()
+	RegisterSignal(quirk_holder, COMSIG_LIVING_MOB_BUMP, PROC_REF(on_bump))
+
+/datum/quirk/defensive/proc/on_bump(mob/living/bumped)
+	if(!COOLDOWN_FINISHED(src, shovingspam))
```

Comment:
This should use the same cooldown that manual shoving uses or this could be abused

Reply:
Cooldown is more for the fact it auto shoves and runs the proc fast. It already has the shoving cooldown but it shouldn't be as fast as shoving